.github/scripts/otp-compliance.es

```diff
@@ -2384,6 +2411,274 @@ run_openvex1(VexStmts, VexTableFile, Branch, VexPath) ->
     Statements = calculate_statements(VexStmts, VexTableFile, Branch, VexPath),
     lists:foreach(fun (St) -> io:format("~ts", [St]) end, Statements).
 
+verify_openvex(#{branch := Branch, vex_path := VexPath}) ->
+    UpdatedBranch = maint_to_otp_conversion(Branch),
+    OpenVEX = read_openvex(VexPath, UpdatedBranch),
+    Advisory = download_advisory_from_branch(UpdatedBranch),
+    case verify_advisory_against_openvex(OpenVEX, Advisory) of
+        [] ->
+            ok;
+        MissingAdvisories when is_list(MissingAdvisories) ->
+            create_advisory(MissingAdvisories)
+    end.
+
+read_openvex(VexPath, Branch) ->
+    InitVex = fetch_openvex_filename(Branch, VexPath),
+    case filelib:is_file(InitVex) of
+        true -> % file exists
+            decode(InitVex);
+        false -> % create file
+            throw(file_not_found)
+    end.
+
+create_advisory(Advisories) ->
+    io:format("Missing:~n~p~n~n", [Advisories]),
+    throw(no_advisory_created).
+
+generate_gh_link(Part) ->
+    "\"/repos/erlang/otp/security-advisories?" ++ Part ++ "\"".
+
+download_advisory_from_branch(Branch) ->
+    Cmd = generate_gh_link("state=published&direction=desc&per_page=100&sort=updated"),
+    paginate_years(Branch, Cmd).
+
+paginate_years(Branch, Cmd) when is_binary(Cmd) ->
+    paginate_years(Branch, erlang:binary_to_list(Cmd));
+paginate_years(Branch, Cmd) when is_list(Cmd) ->
+    Cmd0 = "gh api -i -H \"Accept: application/vnd.github+json\" -H \"X-GitHub-Api-Version: 2022-11-28\" ",
+    Cmd1 = Cmd0 ++ Cmd,
+    io:format("~p~n", [Cmd1]),
+    AdvisoryStr = os:cmd(Cmd1, #{ exception_on_failure => true }),
+    UnicodeBin = unicode:characters_to_binary(AdvisoryStr),
+    RawHTTP = string:split(UnicodeBin, "\n", all),
+    Body0 = extract_http_gh_body(RawHTTP),
+    {{LowRangeYear, _Month, _Day}, _} = calendar:local_time(),
+
+    %% Get the latest 5 years of CVEs
```

Comment:
we only maintain releases for 3 years, but I thought it is better to be cautious.

Reply:
the script later on will filter CVEs who are not part of the latest 3 releases.